codalab/lib/bundle_store.py

```diff
@@ -383,3 +395,130 @@ def cleanup(self, uuid, dry_run):
+                bundle_model = db_bundle_by_uuid.get(uuid, None)
+                if bundle_model == None:
+                    to_delete += [bundle_path]
+                    continue
+                # Delete dependencies stored inside of READY or FAILED bundles
+                if bundle_model.state in [State.READY, State.FAILED]:
+                    dep_paths = [
+                            os.path.join(bundle_path, dep.child_path)
+                            for dep in bundle_model.dependencies
+                          ]
+                    to_delete += filter(os.path.exists, dep_paths)
+            return to_delete
+
+        def _check_other_paths(other_paths, db_bundle_by_uuid):
+            """
+            Given a list of non-bundle paths, returns a list of paths to delete.
+            """
+            to_delete = []
+            for path in other_paths:
+                uuid = _get_uuid(path)
+                bundle_model = db_bundle_by_uuid.get(uuid, None)
+                if bundle_model == None:
+                    to_delete += [path]
+                    continue
+                ends_with_ext = path.endswith('.cid') or path.endswith('.status') or path.endswith('.sh')
+                if bundle_model.state in [State.READY, State.FAILED]:
+                    if ends_with_ext:
+                        to_delete += [path]
+                        continue
+                    elif '.' in path:
+                        print >> sys.stderr, 'WARNING: File %s is likely junk.' % path
+            return to_delete
+
+
+        partitions, _ = path_util.ls(self.partitions)
+        trash_count = 0
+
+        for partition in partitions:
+            partition_path = os.path.join(self.partitions, partition, MultiDiskBundleStore.DATA_SUBDIRECTORY)
+            entries = map(lambda f: os.path.join(partition_path, f),
+                          reduce(lambda d,f: d + f, path_util.ls(partition_path)))
+            bundle_paths = filter(_is_bundle, entries)
+            other_paths = set(entries) - set(bundle_paths)
+
+            uuids = map(_get_uuid, bundle_paths)
+            db_bundles = model.batch_get_bundles(uuid=uuids)
+            db_bundle_by_uuid = dict()
+            for bundle in db_bundles:
+                db_bundle_by_uuid[bundle.uuid] = bundle
+
+            # Check both bundles and non-bundles and remove each
+            for to_delete in _check_bundle_paths(bundle_paths, db_bundle_by_uuid):
+                trash_count += 1
+                _delete_path(to_delete)
+            for to_delete in _check_other_paths(other_paths, db_bundle_by_uuid):
+                trash_count += 1
+                _delete_path(to_delete)
+
+
+            # Check for each bundle if we need to compute its data_hash
+            data_hash_recomputed = 0
+
+            for bundle in bundle_paths:
+                uuid = _get_uuid(bundle)
+                bundle_model = db_bundle_by_uuid[uuid]
```

Comment:
Yeah that was my mistake, I forgot that bundles not in the db will not necessarily have been deleted by this line, only if the -f flag had been used. Should be fixed now.